File: src/main/java/dev/losterixx/sCore/paper/features/warps/SetWarpCommand.java
package dev.losterixx.sCore.paper.features.warps;

import dev.dejvokep.boostedyaml.YamlDocument;
import dev.losterixx.sCore.paper.PaperMain;
import dev.losterixx.sCore.paper.utils.ConfigManager;
import net.kyori.adventure.text.Component;
import net.kyori.adventure.text.minimessage.MiniMessage;
import org.bukkit.Location;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

public class SetWarpCommand implements CommandExecutor {

    private MiniMessage mm = PaperMain.mm;
    private PaperMain main = PaperMain.getInstance();
    private ConfigManager configManager = main.getConfigManager();
    private YamlDocument getConfig() { return configManager.getConfig("config"); }
    private YamlDocument getMessages() { return configManager.getConfig("messages"); }
    private YamlDocument getData() { return configManager.getConfig("data"); }
    private Component getPrefix() { return mm.deserialize(getConfig().getString("prefix")); }

    @Override
    public boolean onCommand(CommandSender sender, Command command, String label, String[] args) {

        if (!(sender instanceof Player)) {
            sender.sendMessage(getPrefix().append(mm.deserialize(getMessages().getString("general.noPlayer"))));
            return false;
        }

        if (!sender.hasPermission("sCore.command.setwarp")) {
            sender.sendMessage(getPrefix().append(mm.deserialize(getMessages().getString("general.noPerms"))));
            return false;
        }

        if (args.length != 1) {
            sender.sendMessage(getPrefix().append(mm.deserialize(getMessages().getString("commands.setwarp.usage"))));
            return false;
        }

        String warpName = args[0];

        if (getData().contains("warps." + warpName)) {
            sender.sendMessage(getPrefix().append(mm.deserialize(getMessages().getString("commands.setwarp.alreadExists").replaceAll("%warp%", warpName))));
            return false;
        }

        Player player = (Player) sender;
        Location loc = player.getLocation();

        getData().set("warps." + warpName + ".world", loc.getWorld().getName());
        getData().set("warps." + warpName + ".x", loc.getX());
        getData().set("warps." + warpName + ".y", loc.getY());
        getData().set("warps." + warpName + ".z", loc.getZ());
        getData().set("warps." + warpName + ".yaw", loc.getYaw());
        getData().set("warps." + warpName + ".pitch", loc.getPitch());

        configManager.saveConfig("data");

        player.sendMessage(getPrefix().append(mm.deserialize(getMessages().getString("commands.setwarp.set").replaceAll("%warp%", warpName))));

        return false;
    }

}

File: src/main/kotlin/dev/losterixx/sCore/features/warps/SetWarpCommand.kt
package dev.losterixx.sCore.features.warps

import dev.losterixx.sCore.Main
import dev.losterixx.sCore.utils.ConfigManager
import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player

class SetWarpCommand : CommandExecutor {

    private val mm = Main.miniMessage
    private val main = Main.instance
    private fun getConfig() = ConfigManager.getConfig("config")
    private fun getMessages() = ConfigManager.getConfig(getConfig().getString("langFile", "english"))
    private fun getData() = ConfigManager.getConfig("data")
    private fun getPrefix() = getConfig().getString("prefix") ?: Main.DEFAULT_PREFIX

    override fun onCommand(sender: CommandSender, command: Command, label: String, args: Array<out String>): Boolean {

        if (sender !is Player) {
            sender.sendMessage(mm.deserialize(getPrefix() + getMessages().getString("general.noPlayer")))
            return false
        }

        if (!sender.hasPermission("sCore.command.setwarp")) {
            sender.sendMessage(mm.deserialize(getPrefix() + getMessages().getString("general.noPerms")))
            return false
        }

        if (args.size != 1) {
            sender.sendMessage(mm.deserialize(getPrefix() + getMessages().getString("commands.setwarp.usage")))
            return false
        }

        val warpName = args[0]

        if (getData().contains("warps.$warpName")) {
            sender.sendMessage(mm.deserialize(getPrefix() + getMessages().getString("commands.setwarp.alreadyExists")
                .replace("%warp%", warpName)))
            return false
        }

        val loc = sender.location

        getData().set("warps.$warpName.world", loc.world?.name)
        getData().set("warps.$warpName.x", loc.x)
        getData().set("warps.$warpName.y", loc.y)
        getData().set("warps.$warpName.z", loc.z)
        getData().set("warps.$warpName.yaw", loc.yaw)
        getData().set("warps.$warpName.pitch", loc.pitch)
        ConfigManager.saveConfig("data")

        sender.sendMessage(mm.deserialize(getPrefix() + getMessages().getString("commands.setwarp.set")
            .replace("%warp%", warpName)))

        return false
    }

}